docs/components-upgrade.md

```diff
@@ -0,0 +1,80 @@
+# Upgrading an environment
+
+The ACS installation is made of several components among which:
+
+ - Content service
+ - share
+ - digital workspace
+ - Search service
+ - ...
+
+## Pre-requisites
+
+Before proceeding to an upgrade, an administrator needs to:
+
+ * Make sure the [upgrade path and documentation](https://docs.alfresco.com/content-services/latest/upgrade/) is followed
+
+ * Appropriate backups of database, contentstore and indexes are done and can be restored in case a rollback is needed
+
+ * Initial deployment must have been done using this Ansible playbook
+
+## Supported Components
+
+### Content service
+
+#### Upgrade types
+
+There are different kinds of upgrade one may want to proceed. For example a major upgrade would be upgrading from 7.0.x to 7.1.x.
+A minor upgrade (or service pack upgrade) would be moving from 7.0.0 to 7.0.1.
+Both major and minor upgrades require setting up a new environment and porting data and customizations from the source environment to the target one. The main reason behind this limitation is that major upgrades usually come with underlying software stack changes.
+These migrations cannot be done "in-place".
+
+Alfresco also releases some hotfixes and an hotfix upgrade would be moving from 7.0.1 to 7.0.1.4. This type of upgrade can be done "in-place" and is supported by the playbook.
+
+> Please note that "in-place" upgrade still need to match the upgrade pre-requisites
+
+#### Proceeding to an "in-place" hotfix upgrade
+
+In order to apply a later hotfix, you need to first match the pre-requisites, then change the ACS version to point to the hotfix version in the appropriate file, and finally run the playbook again.
+
+In the example bellow we want to upgrade from the initial 7.0.1 installtion to 7.0.1.4 hotfix:
+
+Edit `7.0.N-extra-vars.yml` and changes the the bellow snippet
+
+```
+---           
+acs:         
+  version: 7.0.1
+  edition: Enterprise 
+...
+```
+
+to
+
+```
+---           
+acs:         
+  version: 7.0.1.4
+  edition: Enterprise 
+...
+```
+
+> IMPORTANT: make sure you do not set the version to a version number that's not a hotfix (version number needs to be 4 digits and the 3 first ones needs to match the ones of the initially deployed version)
```

Comment:
what are the side effects if someone doesn't read in time this important note?

Maybe isn't trivial to check semantically that the new version is not an in-place upgrade, but I am asking just to be sure that for any chance this has been already discussed elsewhere

Reply:
That's a valid concern. People could end up trying an upgrade that's not supported.
E.g. if they try a service pack upgrade, that may go well, but in case of major upgrade chances are that would break the installation.
Thinking of it maybe we could add a file on each target host which describe which version of the components have been deployed locally on the host together with an additional setting to allow for upgrades... That would allow to:
* make sure people know what they are doing
* check the version match what was previously installed

What do you think?

Reply:
I would implement it, even in most simple way, to avoid unintended disruptions (if versionInstalled != desiredVersion throw error), the override can just be to manually update the guard file with the desiredVersion (don't know if it make sense with Ansible, maybe in a couple of months I will understand it lol)